### ISSUE TITLE
Upload multiple files. Variables generated format

### ISSUE THREAD

**Issue by alobanov** [apollographql/apollo-ios#979]
Apollo supported [GraphQL multipart request specification](https://github.com/jaydenseric/graphql-multipart-request-spec#multipart-form-field-structure), but I can't do something like this:

```swift
// type ImageInput = Type: String!, file: Upload!
mutation SaveFiles($images: [ImageInput]!) { 
  saveFiles(images: $images) {
    ... on Result { ... }
    ... on ValidationErrorsList { ... }
  }
}
```

Implementation in project:
```
let image1 = ImageInput(type: .idCardFront, file: "0")
let image2 = ImageInput(type: .idCardBack, file: "1")

let img = (UIImage(named: "image")?.pngData())!
let file1 = GraphQLFile(fieldName: "images", originalName: "ImageFront.png", mimeType: "image/png", data: img)
let file2 = GraphQLFile(fieldName: "images", originalName: "ImageBack.png", mimeType: "image/png", data: img)

let mutation = SaveFiles(...)
apollo.upload(operation: mutation, files: [file1, file2])
```

After that, under the hud variables transform to:
```swift
// Apollo generate:
"variables": { "images": [null, null] }

// !!! but it should be:
"variables": { "images": [ {"Type": "idCardFront", file: null}, {"Type": "idCardBack", file: null} }
```

Under the hud file variable transform to: ([see the code](https://github.com/apollographql/apollo-ios/blob/4f3e862a0fc62790c9d94d17a646d8febb054f6f/Sources/Apollo/RequestCreator.swift#L127))
```swift
// Apollo generate:
map = {"0": "variable.images.0", "1": "variable.images.1"}

// But it should looks like:
map = {"0": "variable.images.0.file", "1": "variable.images.1.file"}
```
---
Unfortunately, I can't ask my server side to change implementation. Can you help me with this?

**Comment by yamaha252** [comment]
The same for me.
If I use [apollo-upload-client](https://github.com/jaydenseric/apollo-upload-client) which implements the same specification for web, it works in a properly way. I can use Upload type everywhere I want (inside deep objects, arrays etc) but here iOS implementation is quite strange. It's impossible to use multiple file fields (which aren't an array).

**Comment by designatednerd** [comment]
Hey all, yeah, I realize it's a bit strange. I'm hoping to get a better look at special-casing codegen for `Upload` types with the Swift Codegen rewrite to fix this permanently, but that's going to take a while. 

In the meantime, I would be more than happy to take a look at a pull request on existing stuff making this better.

**Comment by designatednerd** [comment]
Please see #1279 for what should be a solution to this.

**Comment by designatednerd** [comment]
This fix has shipped with `0.29.0`. If you are still having problems after upgrading to that version, please open a new issue with details, noting that you are using that version. Thank you!
